bin/middlewares/map_style.js

```diff
@@ -0,0 +1,40 @@
+const styleBuilder = require('@qwant/map-style-builder')
+const styleConfigure = require('mapbox_style_configure')
+const qwantStyle = require('@qwant/qwant-basic-gl-style/style.json')
+const path = require('path')
+
+
+module.exports = function (config, languages) {
+
+  /* pre-build style on server start */
+  let options = {
+    output: 'production', // 'debug' | 'production' | 'omt'
+    outPath : __dirname + '/../public/mapstyle',
+    i18n : true,
+    icons : true,
+    pixelRatios : [1,2],
+    styleDir : path.resolve(path.join(__dirname, '..', '..', 'node_modules', '@qwant' , 'qwant-basic-gl-style')),
+    conf : {tileserver_base : '{tileserver_base}', tileserver_poi : '{tileserver_poi}', spriteserver : '{spriteserver}', fontserver : '{fontserver}'},
+    ignoreSprite : true
+  }
+
+  let builtStyle = styleBuilder(qwantStyle, options)
+
+  return function (req, res) {
+    /* get lang routine */
+    let langParameter = req.query.lang
+    let matchedLanguage = languages.defaultLanguage
+    if(langParameter) {
+      matchedLanguage = languages.supportedLanguages.find((supportedLanguage) => supportedLanguage.code === langParameter)
+      if(!matchedLanguage) {
+        res.send(400)
```

Comment:
`res.send` with a status code is deprecated.  
And it is actually an error to call `res.setHeader` after `res.send` so we should return here.  
Maybe we could include an explicit json error in the response too.

Reply:
i'll use res.status
update to a json like {error : Unsupported language code : #lang code}
and return to avoid another send :)